### PR TITLE
Added support for running CPD reports

### PR DESCRIPTION
**Pull request by rahulsom**
CPD support only exists for Objective C, but that's better than no support. I believe this leaves a lot to be desired configurability-wise. Please let me know if you like the general premise, and then I can work towards improving this.

### CODE REVIEW

**Comment by renep** [comment]
In general I wanted to add analytics stuff for quite some time. I'm wondering what would be the best tools to use. I for example think of starting with OCLint.
I have to try the OCLint stuff first with my projects to get a feel, and I think I will also try this pull request. (I hope I find the time soon ;) )

**Comment by rahulsom** [comment]
CPD is a simple Copy Paste Detector. I believe it can stay orthogonal to OCLint. I'm using OCLint as well in my project. However since this plugin cannot (yet) generate json compilation database, I'm using xctool for that one reason.
If I can, I'll try to generate the json database within this plugin.

**Comment by renep** [comment]
The json compilation database should be an easy task to create, because the xcodebuild output is stored in the xcodebuild-output.txt in the build folder.
I think that the OCLint stuff could be quite easy so include.  I have tried the command line tools and have not yet achieved it to create a nice report. I think I have missed something.

My opinion for now is that I want to try your pull request and I think I will add this soon into the development branch.

One thing that I find important and very helpful are unit test. This plugin has already over 200, so it would be nice to have tests for every new stuff.

**Comment by rahulsom** [comment]
Thanks! I'll work on unit tests

File: plugin/src/test/groovy/org/openbakery/cpd/CpdTaskTest.groovy [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+                src: 'http://tcpdiag.dl.sourceforge.net/project/pmd/pmd/4.2.5/pmd-bin-4.2.5.zip',
+                dest: new File("${destDir}/pmd-bin-4.2.5.zip")
+        ]).times(1).returns()
+        antMock.get([
+                src: 'https://raw.githubusercontent.com/jkennedy1980/Objective-C-CPD-Language/master/releases/ObjCLanguage-0.0.7-SNAPSHOT.jar',
+                dest: new File("${destDir}/ObjCLanguage-0.0.7-SNAPSHOT.jar")
+        ]).times(1)
+        antMock.unzip(
+                src: "${destDir}/pmd-bin-4.2.5.zip", dest: "${destDir}/tools", overwrite: 'true'
+        ).times(1)
```
Comment (rahulsom) [inline plugin/src/test/groovy/org/openbakery/cpd/CpdTaskTest.groovy]: @renep I'm having trouble hooking a closure to execute with an expectation. It looks like GMock doesn't allow for that.
Have you done something similar?

Comment (renep) [inline plugin/src/test/groovy/org/openbakery/cpd/CpdTaskTest.groovy]: Take a look here this should help:
https://github.com/openbakery/gradle-xcodePlugin/blob/develop/plugin/src/test/groovy/org/openbakery/oclint/OCLintTaskTest.groovy

**Comment by renep** [comment]
Merge to develop branch

**Comment by renep** [comment]
If you like I can provide a develop build for testing.

**Comment by rahulsom** [comment]
Thanks! I'll pull from develop and create a local install.

**Pull request closed without merging**